tests/unit/common/test_resource.py

```diff
@@ -142,3 +142,79 @@ def test_check_ascii_256(self):
+                'OC_RESOURCE_TYPE': 'opencensus.io/example',
+                'OC_RESOURCE_LABELS': 'k1=v1,k2=v2'
+        }):
+            resource = resource_module.get_from_env()
+        self.assertEqual(resource.type, 'opencensus.io/example')
+        self.assertDictEqual(resource.labels, {'k1': 'v1', 'k2': 'v2'})
+
+    def test_get_from_env_no_type(self):
+        with mock.patch.dict('os.environ', {
+                'OC_RESOURCE_LABELS': 'k1=v1,k2=v2'
+        }):
+            self.assertIsNone(resource_module.get_from_env())
+
+    def test_get_from_env_no_labels(self):
+        with mock.patch.dict('os.environ', {
+                'OC_RESOURCE_TYPE': 'opencensus.io/example',
+        }):
+            resource = resource_module.get_from_env()
+        self.assertEqual(resource.type, 'opencensus.io/example')
+        self.assertDictEqual(resource.labels, {})
+
+    def test_get_from_env_outer_spaces(self):
+        with mock.patch.dict('os.environ', {
+                'OC_RESOURCE_TYPE': ' opencensus.io/example  ',
+                'OC_RESOURCE_LABELS': 'k1= v1 ,  k2=v2  '
+        }):
+            resource = resource_module.get_from_env()
+        self.assertEqual(resource.type, 'opencensus.io/example')
+        self.assertDictEqual(resource.labels, {'k1': 'v1', 'k2': 'v2'})
+
+    def test_get_from_env_inner_spaces(self):
+        # Spaces inside key/label values should be contained by quotes, refuse
+        # to parse this.
+        with mock.patch.dict('os.environ', {
+                'OC_RESOURCE_TYPE': 'opencensus.io / example',
+                'OC_RESOURCE_LABELS': 'key one=value one,  key two= value two'
+        }):
+            resource = resource_module.get_from_env()
+        self.assertEqual(resource.type, 'opencensus.io / example')
+        self.assertDictEqual(resource.labels, {})
+
+    def test_get_from_env_quoted_chars(self):
+        with mock.patch.dict('os.environ', {
+                'OC_RESOURCE_TYPE': 'opencensus.io/example',
+                'OC_RESOURCE_LABELS': '"k1=\'"="v1,,,", "k2"=\'="=??\''
+        }):
+            resource = resource_module.get_from_env()
+        self.assertDictEqual(resource.labels, {"k1='": 'v1,,,', 'k2': '="=??'})
+
+    def test_parse_labels(self):
+        self.assertEqual(resource_module.parse_labels("k1=v1"), {'k1': 'v1'})
+        self.assertEqual(
+            resource_module.parse_labels("k1=v1,k2=v2"),
+            {'k1': 'v1', 'k2': 'v2'})
+
+        self.assertEqual(
+            resource_module.parse_labels('k1="v1"'),
+            {'k1': 'v1'})
+
+        self.assertEqual(
+            resource_module.parse_labels('k1="v1==,"'),
+            {'k1': 'v1==,'})
+
+        self.assertEqual(
+            resource_module.parse_labels('k1="one/two/three"'),
+            {'k1': 'one/two/three'})
+
+        self.assertEqual(
+            resource_module.parse_labels('k1="one\\two\\three"'),
+            {'k1': 'one\\two\\three'})
+
+        with self.assertRaises(ValueError):
```

Comment:
Please fix this.

Reply:
Sorry @songy23! That's what I get for not rerunning the tests...